File: productCommunityService/src/main/java/com/nagarro/productCommunitySystem/service/ReviewService.java
package com.nagarro.productCommunitySystem.service;

import com.nagarro.productCommunitySystem.model.Product;
import com.nagarro.productCommunitySystem.model.Review;

import java.util.List;

public interface ReviewService {

    void addReview(long productId, Review review) throws Exception;

    void approveReview(long reviewId) throws Exception;

    void unApproveReview(long reviewId) throws Exception;
    Review requestReview(Product product) throws Exception;
    int getReviewsCount() throws Exception;

    List<Review> getReviews() throws Exception;
}

File: ProductCommunityWeb/src/app/services/review.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  constructor(private http: HttpClient) { }

  addReview(productId: number, review: any) {
    return this.http.post("http://localhost:8080/api/review/" + productId, review);
  }

  approveReview(reviewId: number) {
    return this.http.get("http://localhost:8080/api/review/approve/" + reviewId);
  }

  unApproveReview(reviewId: number) {
    return this.http.get("http://localhost:8080/api/review/unapprove/" + reviewId);
  }

  getReviewCount() {
    return this.http.get("http://localhost:8080/api/review/count");
  }

  getReviews() {
    return this.http.get("http://localhost:8080/api/review");
  }

}
